graspologic/simulations/simulations.py

```diff
@@ -795,3 +796,184 @@ def p_from_latent(X, Y=None, rescale=False, loops=True):
+    which indicates the probability of connection between nodes based upon
+    their community membership.
+    Each node is assigned a membership vector drawn from Dirichlet distribution
+    with parameter :math:`\vec{\alpha}`. The entries of this vector indicate the
+    probabilities for that node of pertaining to each community when interacting with
+    another node. Each node's membership is determined according to those probabilities.
+    Finally, interactions are sampled according to the assigned memberships.
+
+    Read more in the :ref:`tutorials <simulations_tutorials>`
+
+    Parameters
+    ----------
+    n: int
+        Number of vertices of the graph.
+
+    p: array-like, shape (n_communities, n_communities)
+        Probability of an edge between each of the communities, where ``p[i, j]``
+        indicates the probability of a connection between edges in communities
+        :math:`(i, j)`.
+        0 < ``p[i, j]`` < 1 for all :math:`i, j`.
+
+    alpha: array-like, shape (n_communities,)
+        Parameter alpha of the Dirichlet distribution used
+        to sample the mixed-membership vectors for each node.
+        ``alpha[i]`` > 0 for all :math:`i`.
+
+    rng: numpy.random.Generator, optional (default = None)
+        :class:`numpy.random.Generator` object to sample from distributions.
+        If None, the random number generator is the Generator object constructed
+        by ``np.random.default_rng()``.
+
+    directed: boolean, optional (default=False)
+        If False, output adjacency matrix will be symmetric. Otherwise, output adjacency
+        matrix will be asymmetric.
+
+    loops: boolean, optional (default=False)
+        If False, no edges will be sampled in the diagonal. Otherwise, edges
+        are sampled in the diagonal.
+
+    return_labels: boolean, optional (default=False)
+        If False, the only output is the adjacency matrix.
+        If True, output is a tuple. The first element of the tuple is the adjacency
+        matrix. The second element is a matrix in which the :math:`(i^{th}, j^{th})`
+        entry indicates the membership assigned to node i when interacting with node j.
+        Community 1 is labeled with a 0, community 2 with 1, etc.
+        -1 indicates that no community was assigned for that interaction.
+
+    References
+    ----------
+    .. [1] Airoldi, Edoardo, et al. “Mixed Membership Stochastic Blockmodels.”
+       Journal of Machine Learning Research, vol. 9, 2008, pp. 1981–2014.
+
+    Returns
+    -------
+    A: ndarray, shape (n, n)
+        Sampled adjacency matrix
+    labels: ndarray, shape (n, n), optional
+        Array containing the membership assigned to each node when interacting with
+        another node.
+
+    Examples
+    --------
+    >>> rng = np.random.default_rng(1)
+    >>> np.random.seed(1)
+    >>> n = 6
+    >>> p = [[0.5, 0], [0, 1]]
+
+    To sample a binary MMSBM in which all nodes pertain to community two:
```

Comment:
is there even a small chance of this not being true given \alpha?

Reply:
There is technically a very small probability but it is very very small. Because when for example alpha = [1000 1] all membership vectors will look something similar to [0.998 0.002] or analogous therefore when sampling the membership for each interaction using multinomial they should belong to that community 99.8% of the time. Obviously, the sampling methods in numpy may cause some variation from the theoretical case depending on initialization of seed etc. but given such high probabilities of belonging to a single class the variation should be fairly small.

I will write a quick test confirming this using ER. I believe that generating an SBM from same block connectivity Matrix should also work despite the different number of nodes.

Reply:
i just meant purely in terms of the language here - maybe say "all nodes are very likely to pertain to..." as opposed to making this certain, cause that just tripped me up here.

Reply:
Yes, I totally agree and I appreciate you for pointing it out! Will fix both in docs and in tests.